### DISCUSSION TITLE
Retrieving polyhedral surface from 3D mesh

### DISCUSSION THREAD

**Discussion by mjziebarth** [CGAL/cgal#6804] [Q&A] [answered]
Dear all,
thanks for the great work with CGAL!

I am using CGAL to generate a 3D mesh for a geophysical model. So far, I have followed example [4.3.2 Remeshing a Polyhedral Domain with Surfaces](https://doc.cgal.org/latest/Mesh_3/index.html#title24), which is quite similar in structure to what I need, and
I have been able to construct the mesh (once I fixed all issues in describing the polyhedral setup, CGAL meshing worked like a charm!).

Now I am wondering whether there is any standard way to retrieve the facets of the input polyhedra within the output mesh. I need to know these facets to impose boundary conditions in a finite element analysis that I create the mesh for. I guess that I should be able to figure out the corresponding facets brute force from the output mesh since I know the input polyhedra. However, if there was a standard way to do so within the meshing process, for instance a vertex marker or something similar, I would greatly prefer that.

So far, I have not found a method like this in the documentation and hence I wanted to ask: is there a standard method to keep track of input vertices in the `CGAL::make_mesh_3` approach?

Thanks a lot for the work put into the library and I'd be happy to learn if such a feature exists (even if that just means that I have to figure out the facets after mesh generation).

Best
Malte

**Comment by sloriot** [comment, marked as answer]
Hi,

check [`facets_in_complex_begin()`](https://doc.cgal.org/latest/Mesh_3/classMeshComplex__3InTriangulation__3.html#ad3e6508fc82806e1001957d35452cffc)

**Reply by mjziebarth** [reply]
Thanks a lot, that seems to have solved my problem at hand.

I still needed a bit of bookkeeping to reassign the facets in the complex to the input surfaces
(maybe I am not using the subclassing to its capability) but having an iterator
over the facets has proved very useful.
